### PR TITLE
avm2: Implement more event classes

### PR DESCRIPTION
**Pull request by danielhjacobs**
Implement the methods and constants for ErrorEvent, TextEvent, and SecurityErrorEvent and update IOErrorEvent to use proper inheritance. Clashes with #7013, as I started work on it before I noticed that implemented "text" on IOErrorEvent.

### CODE REVIEW

**Comment by Herschel** [comment]
Just a note for myself in the future that a lot of these things are defined slightly differently, e.g. `ErrorEvent.errorId` is a getter with a private property:
```
      private var m_errorID:int;      
      public function get errorID() : int
      {
         return this.m_errorID;
      }
```
Specifically it's a normal property on the object and not marked `native`, in contrast to most of the `Event` and `MouseEvent` properties. So eventually `errorID` shouldn't even be part of `EventData`, I think.

But for now this isn't of any practical consequence and I expect most of this to get cleaned up when we start generating from SWC.

File: core/src/avm2/events.rs [hunk 1]
```diff
@@ -87,6 +87,9 @@ impl KeyModifiers {
 #[collect(no_drop)]
 pub enum EventData<'gc> {
     Empty,
+    Error {
```
Comment (kmeisthax) [inline core/src/avm2/events.rs:90]: `ErrorEvent` subclasses `TextEvent`, so `EventData::Error` should have all the fields in `EventData::Text`.

File: core/src/avm2/globals/flash/events/textevent.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+/// Implements `text`'s getter.
+fn text<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Some(evt) = this.as_event() {
+            if let EventData::Text { text, .. } = evt.event_data() {
```
Comment (kmeisthax) [inline core/src/avm2/globals/flash/events/textevent.rs:68]: Subclasses of `TextEvent` will also have a `text` property; this currently will return `undefined` for all subclasses that have a different native `Error` type.

**Review by kmeisthax — COMMENTED** [review]
Yeah, related to that; I'm actually getting the feeling that the whole `Event` enum might have been a bad idea on my part - it makes inherited properties far harder to implement, as I've noted in the comments below.

File: core/src/avm2/globals/flash/events/textevent.rs (hunk 2 above)
Comment (danielhjacobs) [reply core/src/avm2/globals/flash/events/textevent.rs:68]: This seems complicated to fix. Is the way to fix it without removing the enum to add the same method for each inherited class? And also adding IOErrorEvent and SecurityErrorEvent to the EventData enum with the exact same attributes as ErrorEvent (which, as you noted,  needs TextEvent's attributes added to it)?

**Comment by danielhjacobs** [comment]
If you want to remove the EventData enum from current classes I'd be fine rebasing. Removing it properly is beyond my level of knowledge (most of this PR was finding the right things to copy and paste. Otherwise I'll try to work around the enum issues tomorrow.

**Comment by danielhjacobs** [comment]
I really messed up the history for this commit. If it's to be merged, probably best as a squash and merge.

**Comment by danielhjacobs** [comment]
Replaced with #7037 since I messed up the commit history. I believe collaborators can squash and merge, but it's easier for me to work with a normal branch as opposed to a messed up branch if I need to make any changes, and I couldn't figure out how to properly rebase this branch without creating merge conflicts.

**Pull request closed without merging**